### PR TITLE
Migrations for rustc_expand transcribe.rs

### PR DESCRIPTION
**Pull request by nidnogg**
This PR includes some migrations to the new diagnostics API for the `rustc_expand` module.
r? @davidtwco

### CODE REVIEW

File: compiler/rustc_error_messages/locales/en-US/expand.ftl
Comment (compiler-errors) [inline compiler/rustc_error_messages/locales/en-US/expand.ftl:14]: @davidtwco I wonder if we want to make `x.py test tidy` lint on trailing newline in fluent files

File: compiler/rustc_expand/src/mbe/transcribe.rs
Comment (davidtwco) [inline compiler/rustc_expand/src/mbe/transcribe.rs:69]: Could you move the diagnostic structs to `rustc_expand::errors`? That's what we've been doing for every other crate so that the errors are off to the side.

File: compiler/rustc_error_messages/locales/en-US/expand.ftl
Comment (davidtwco) [reply compiler/rustc_error_messages/locales/en-US/expand.ftl:14]: We certainly could, it's an easy task for someone to do at some point.

**Review by davidtwco — COMMENTED** [review]
Looking good so far, will wait until you're happy to un-draft this to merge.

File: compiler/rustc_error_messages/locales/en-US/expand.ftl
Comment (Xiretza) [reply compiler/rustc_error_messages/locales/en-US/expand.ftl:14]: I've opened #100671.

File: compiler/rustc_error_messages/locales/en-US/expand.ftl
Comment (davidtwco) [inline compiler/rustc_error_messages/locales/en-US/expand.ftl:13]: Oh, and the slugs here should start with `expand_` in this file.

File: compiler/rustc_expand/src/mbe/transcribe.rs
Comment (nidnogg) [reply compiler/rustc_expand/src/mbe/transcribe.rs:69]: Solved in the upcoming commit

File: compiler/rustc_error_messages/locales/en-US/expand.ftl
Comment (nidnogg) [reply compiler/rustc_error_messages/locales/en-US/expand.ftl:13]: Ops, my bad. Solved in the latest commit.

File: compiler/rustc_expand/src/errors.rs
Comment (davidtwco) [inline compiler/rustc_expand/src/errors.rs:43]: This is the wrong slug for the error, which is causing your test failure - unfortunately we don't have compile-time checking that the fields match up with the interpolated variables yet.

**Review by davidtwco — APPROVED** [review]

**Pull request merged**
